### ISSUE TITLE
it can never be simple. npm install fails.

### ISSUE THREAD

**Issue by andrew-luhring** [pilwon/ultimate-seed#107]
## npm install fails with:

``` sh
npm prepublish script executed
npm ERR! peerinvalid The package karma does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer grunt-karma@0.6.2 wants karma@~0.10.0
npm ERR! peerinvalid Peer karma-chrome-launcher@0.1.0 wants karma@>=0.9.3
npm ERR! peerinvalid Peer karma-mocha@0.1.0 wants karma@>=0.9
npm ERR! peerinvalid Peer karma-ng-html2js-preprocessor@0.1.0 wants karma@>=0.9
npm ERR! peerinvalid Peer karma-firefox-launcher@0.1.3 wants karma@>=0.9
npm ERR! peerinvalid Peer karma-html2js-preprocessor@0.1.0 wants karma@>=0.9
npm ERR! peerinvalid Peer karma-jasmine@0.1.5 wants karma@>=0.9
npm ERR! peerinvalid Peer karma-requirejs@0.2.1 wants karma@>=0.9
npm ERR! peerinvalid Peer karma-script-launcher@0.1.0 wants karma@>=0.9
npm ERR! peerinvalid Peer karma-coffee-preprocessor@0.2.1 wants karma@>=0.11.14
npm ERR! peerinvalid Peer karma-phantomjs-launcher@0.1.2 wants karma@>=0.9

npm ERR! System Darwin 13.1.0
npm ERR! command "/usr/local/Cellar/node/0.10.25/bin/node" "/usr/local/bin/npm" "install"
npm ERR! cwd /Users/worker_bee/Sites/ultimate-seed
npm ERR! node -v v0.10.25
npm ERR! npm -v 1.3.24
npm ERR! code EPEERINVALID
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /Users/worker_bee/Sites/ultimate-seed/npm-debug.log
npm ERR! not ok code 0
```

**Comment by yaru22** [comment]
Hi @andrew-luhring, thanks for reporting the error. I updated the karma related dependencies to fix the above error. Please give it another try and let us know if you see any other problems!

**Comment by andrew-luhring** [comment]
@yaru22  Now the error is:

``` bash

Andrews-MacBook-Pro:ultimate-seed worker_bee$ grunt
Running "clean:dev" (clean) task
Cleaning .tmp...OK

Running "browserify2:dev" (browserify2) task
>> Error: module "./bower_components/angular-translate/angular-translate" not found from "/Users/worker_bee/Sites/ultimate-seed/client/js/node_modules/angular.translate.js"
File written to: .tmp/js/main.js
>> Error: module "./bower_components/angular-ui-bootstrap/ui-bootstrap-tpls-0.9.0" not found from "/Users/worker_bee/Sites/ultimate-seed/client/js/node_modules/angular.ui.js"
File written to: .tmp/js/main.js
>> Error: module "./bower_components/angular-ui-router/release/angular-ui-router" not found from "/Users/worker_bee/Sites/ultimate-seed/client/js/node_modules/angular.ui.js"
File written to: .tmp/js/main.js
>> Error: module "./bower_components/bootstrap/dist/js/bootstrap" not found from "/Users/worker_bee/Sites/ultimate-seed/client/js/node_modules/bootstrap.js"
File written to: .tmp/js/main.js
>> Error: module "./bower_components/spinjs/jquery.spin" not found from "/Users/worker_bee/Sites/ultimate-seed/client/js/node_modules/jquery.spin.js"
File written to: .tmp/js/main.js
>> Error: module "./bower_components/restangular/dist/restangular" not found from "/Users/worker_bee/Sites/ultimate-seed/client/js/node_modules/restangular.js"
File written to: .tmp/js/main.js

Running "less:dev" (less) task

Running "copy:dev" (copy) task


Running "symlinkDev" task
Warning: ENOENT, no such file or directory '/Users/worker_bee/Sites/ultimate-seed/client/js/node_modules/bower_components/font-awesome/fonts' Use --force to continue.



```

**Comment by yaru22** [comment]
@andrew-luhring, did you run `bower install`?

**Comment by andrew-luhring** [comment]
ran npm install

**Comment by andrew-luhring** [comment]
no bower install; normally you just do npm install and it installs everything.

**Comment by yaru22** [comment]
@andrew-luhring, you do need to run both `npm install` and `bower install`. `npm install` is for dependencies for the backend and `bower install` is for dependencies for the frontend.

Please see https://github.com/pilwon/ultimate-seed#method-2-git-clone.

Also, regarding your `npm install` failure, could you clone the repo fresh and try it again? We pushed some changes the past few days.

**Comment by pilwon** [comment]
> normally you just do npm install and it installs everything.

@andrew-luhring This statement is incorrect. Some node.js projects are set up this way only because the author set it up like that. We chose not to support chaining `bower install` with `npm install` for finer control. We'd advise you to read through [Installation](https://github.com/pilwon/ultimate-seed#installation) section of README.

**Comment by andrew-luhring** [comment]
Weird.
I know I didn't use bower ever, and I don't think I cloned it... maybe I started it with yeoman.
I'll look at it when I get home.

**Comment by pilwon** [comment]
@andrew-luhring Are you trying it with Yeoman generator? Then it should run both `npm install` and `bower install` with a single command `yo ultimate`. Now that we understand your installation method, the new advice is to either wait for the next Yeoman generator release (it's slightly lagged behind repo's master branch so you might be using an older version) or simply try with @yaru22 's suggestion (cloning the repo).

**Comment by pilwon** [comment]
@andrew-luhring We've just published v0.3.9 to Yeoman generator `generator-ultimate`. Please try again.
